### ISSUE TITLE
Can not use Trainer.test() if train and val dataloaders are not defined

### ISSUE THREAD

**Issue by Varal7** [PyTorchLightning/pytorch-lightning#1857]
## 🐛 Bug

When the model does not define `train_dataloader` and no `val_dataloader`, we can not use `trainer.test(model, test_dataloaders=test_dl)`.
The configuration checks fail with a `MisconfigurationException`.

#### Code sample

```
model = ...  # a model with no `train_dataloader`, `val_dataloader` defined
test_dl = ... # a dataloader
trainer = pl.Trainer()
trainer.test(model, test_dataloaders=test_dl)
```

### Expected behavior

We expect the testing loop to execute.

**Comment by Varal7** [comment]
Whoops, sorry I tried searching for the issue / PR but couldn't find the relevant ones. Looks like there are already good ideas in #1804, #1720, #1754, #1804.

**Comment by williamFalcon** [comment]
Fixed in #1804
